cdap-common/src/main/java/co/cask/cdap/common/namespace/RemoteNamespaceQueryClient.java

```diff
@@ -42,26 +48,41 @@
  */
 public class RemoteNamespaceQueryClient extends AbstractNamespaceQueryClient {
   private final Supplier<EndpointStrategy> endpointStrategySupplier;
+  private final String userId;
 
   @Inject
-  RemoteNamespaceQueryClient(final DiscoveryServiceClient discoveryClient) {
+  RemoteNamespaceQueryClient(final DiscoveryServiceClient discoveryClient, CConfiguration cConf) {
     this.endpointStrategySupplier = Suppliers.memoize(new Supplier<EndpointStrategy>() {
       @Override
       public EndpointStrategy get() {
-        return new RandomEndpointStrategy(discoveryClient.discover(Constants.Service.REMOTE_SYSTEM_OPERATION));
+        return new RandomEndpointStrategy(discoveryClient.discover(Constants.Service.APP_FABRIC_HTTP));
       }
     });
+    try {
+      if (SecurityUtil.isKerberosEnabled(cConf)) {
```

Comment:
@rohitsinha54 two things about BaseHiveExploreService#getHiveDatabase:
1. It should have a version that takes in a ``NamespaceMeta``, instead of just the namespace name. This will be called from ``create``, where you already have the ``NamespaceMeta`` before you call ``getHiveDb``. So in this case you don't need to make a call to ``namespaceQueryAdmin.get()``.
2. For cases other than ``create``, just use the userId from ``AuthenticationContext#getPrincipal()``. 


This needs to be tested thoroughly though.